tests/unit/test_client.py

```diff
@@ -250,6 +250,54 @@ def test__call_api_applying_custom_retry_on_timeout(self):
             [mock.call(foo="bar"), mock.call(foo="bar")],  # was retried once
         )
 
+    def test__call_api_span_creator_called(self):
```

Comment:
Looks like this and the other test should exchange names. This is testing that it's not called.